src/ValueHandler/ProductOptionValueHandler.php

```diff
@@ -161,16 +211,130 @@ public function handle($productVariant, string $optionCode, array $akeneoValue):
         }
     }
 
-    private function isVariantOption(ProductVariantInterface $productVariant, string $attribute): bool
+    private function handleMetricOption(ProductOptionInterface $productOption, string $optionCode, array $akeneoDataValue, ProductInterface $product, ProductVariantInterface $productVariant): void
     {
-        $product = $productVariant->getProduct();
-        Assert::isInstanceOf($product, ProductInterface::class);
-        foreach ($product->getOptions() as $option) {
-            if ($attribute === $option->getCode()) {
-                return true;
+        if (!array_key_exists('amount', $akeneoDataValue)) {
+            throw new LogicException('Amount key not found');
+        }
+        $floatAmount = (string) ($akeneoDataValue['amount']);
+        if (!array_key_exists('unit', $akeneoDataValue)) {
+            throw new LogicException('Unit key not found');
+        }
+        $unit = (string) $akeneoDataValue['unit'];
+        $optionValueCode = $optionCode . '_' . $floatAmount . '_' . $unit;
+
+        $optionValue = $this->getOrCreateProductOptionValue($optionValueCode, $productOption);
+
+        /** @var string[] $locales */
+        $locales = [];
+        if ($this->translationLocaleProvider !== null) {
+            $locales = $this->translationLocaleProvider->getDefinedLocalesCodes();
+        } else {
+            foreach ($product->getTranslations() as $translation) {
+                $locale = $translation->getLocale();
+                if ($locale === null) {
+                    continue;
+                }
+                $locales[] = $locale;
+            }
+        }
```

Comment:
We can extract in a common method to use here and in the handleBooleanOption. Doing so we slightly improve the overall readability.